integration/marathon_test.go

```diff
@@ -1,53 +1,128 @@
 package integration
 
 import (
+	"fmt"
 	"net/http"
+	"os"
 	"time"
 
 	"github.com/containous/traefik/integration/try"
+	"github.com/containous/traefik/types"
+	marathon "github.com/gambol99/go-marathon"
 	"github.com/go-check/check"
 	checker "github.com/vdemeester/shakers"
 )
 
+const (
+	containerNameMesosSlave = "mesos-slave"
+	containerNameMarathon   = "marathon"
+)
+
 // Marathon test suites (using libcompose)
-type MarathonSuite struct{ BaseSuite }
+type MarathonSuite struct {
+	BaseSuite
+	marathonURL string
+}
 
 func (s *MarathonSuite) SetUpSuite(c *check.C) {
 	s.createComposeProject(c, "marathon")
 	s.composeProject.Start(c)
 
-	// FIXME Doesn't work...
-	//// "github.com/gambol99/go-marathon"
-	//config := marathon.NewDefaultConfig()
-	//
-	//marathonClient, err := marathon.NewClient(config)
-	//if err != nil {
-	//	c.Fatalf("Error creating Marathon client. %v", err)
-	//}
-	//
-	//// Wait for Marathon to elect itself leader
-	//err = try.Do(30*time.Second, func() error {
-	//	leader, err := marathonClient.Leader()
-	//
-	//	if err != nil || len(leader) == 0 {
-	//		return fmt.Errorf("Leader not found. %v", err)
-	//	}
-	//
-	//	return nil
-	//})
-	//
-	//c.Assert(err, checker.IsNil)
+	marathonIPAddr := s.composeProject.Container(c, containerNameMarathon).NetworkSettings.IPAddress
+	c.Assert(marathonIPAddr, checker.Not(checker.HasLen), 0)
+	s.marathonURL = "http://" + marathonIPAddr + ":8080"
+
+	// Wait for Marathon readiness prior to creating the client so that we
+	// don't run into the "all cluster members down" state right from the
+	// start.
+	err := try.GetRequest(s.marathonURL+"/v2/leader", 1*time.Minute, try.StatusCodeIs(http.StatusOK))
+	c.Assert(err, checker.IsNil)
+
+	// Add entry for Mesos slave container IP address in the hosts file so
+	// that Traefik can properly forward traffic.
+	// This is necessary as long as we are still using the docker-compose v1
+	// spec. Once we switch to v2 or higher, we can have both the test/builder
+	// container and the Mesos slave container join the same custom network and
+	// enjoy DNS-discoverable container host names.
+	mesosSlaveIPAddr := s.composeProject.Container(c, containerNameMesosSlave).NetworkSettings.IPAddress
+	c.Assert(mesosSlaveIPAddr, checker.Not(checker.HasLen), 0)
+	c.Assert(s.extendDockerHostsFile(containerNameMesosSlave, mesosSlaveIPAddr), checker.IsNil)
```

Comment:
Separating the call and the check would actually be more consistent with our existing style.

Updated.